### ISSUE TITLE
Add a multiple selection system

### ISSUE THREAD

**Issue by alexherbo2** [atlas-engineer/next#405]
Here are some visual examples.

### Download 3-gatsu no Lion episodes from HorribleSubs

[![Download 3-gatsu no Lion episodes from HorribleSubs](https://img.youtube.com/vi_webp/aXaFt75lIqo/maxresdefault.webp)](https://youtu.be/aXaFt75lIqo)

### Extract from Wikipedia the list of cities and towns in Russia

[![Extract from Wikipedia the list of cities and towns in Russia](https://img.youtube.com/vi_webp/PJXCnRBkHDY/maxresdefault.webp)](https://youtu.be/PJXCnRBkHDY)

### Quickly move around a document with no table of contents

[![Quickly move around a document with no table of contents](https://img.youtube.com/vi_webp/gp4_6VGXkOk/maxresdefault.webp)](https://youtu.be/gp4_6VGXkOk)

### Opening links in a paragraph

[![Opening links in a paragraph](https://img.youtube.com/vi_webp/v2Jvk1rhIlc/maxresdefault.webp)](https://youtu.be/v2Jvk1rhIlc)

### Play videos with mpv

[![Play videos with mpv](https://img.youtube.com/vi_webp/gYTi-eXuWdI/maxresdefault.webp)](https://youtu.be/gYTi-eXuWdI)

**Comment by Ambrevar** [comment]
Great samples! :)

Hmm... This makes me realize that this is no more than a cross-bread between link hints and minibuffer multi selection.

In other words: once we have minibuffer multi selection (#395) and overhauled the link hints (#354), we only need to add a "persist hint" action to the link hint commands, bound to, say, `C-x C-s`.

Example:

- C-g 720p M-a C-x C-s

would persist hints on all links matching "720p".

- C-g 720p C-SPACE DOWN C-SPACE C-x C-s

would persist link 1 and 3 matching 720p.

Makes sense?

From there, any hint-based command can reused the currently persisted hints.  Et voilà! :)

**Comment by alexherbo2** [comment]
Not exactly.  Selecting 720p magnet links is actually much more complex.  There is no 720p links.

Here are each step done:

1. Focus a link
2. Create a selection out of the active element
3. Extend selections until hitting the whole list
4. Split selections on each item
5. Click on them to open sub-menus
6. Select sub-entries (480p, 720p, 1080p)
7. Keep selections matching _720p_ in their description
8. Select all links within selections
9. Keep selections matching _Magnet_ in their description
10. Open magnets

https://horriblesubs.info/shows/3-gatsu-no-lion/

**Comment by Ambrevar** [comment]
Then having a "click-all-hints" that simulate a click on all hints would work, wouldn't it?

**Comment by alexherbo2** [comment]
No xD

The problem in that specific example is there is no 720p to click for.  It’s a plain text element.  To download all 720p of the serie we have to select their container and within them, select all Magnet links.

So `click-all-hints` will work, but the issue I faced when I was on the page was to get the links. xD

**Comment by Ambrevar** [comment]
So how did you do it?

**Comment by Ambrevar** [comment]
I mean, how do you define a "selection" precisely?

**Comment by alexherbo2** [comment]
Like [this][Selection].  The selection system is inspired by [Kakoune].

[Selection]: https://github.com/alexherbo2/selection.js
[Kakoune]: http://kakoune.org

**Comment by Ambrevar** [comment]
@jmercouris The title may be misleading, but here "multiple selection" refers to something else.  In particular, most of the examples given in the original post are not possible today with Nyxt.

**Comment by Ambrevar** [comment]
No, there is a small overlap but the original post multiple selection system has some features reminding of Emacs multiple cursor.  It would be useful to have and we have nothing close to this concept.
